packages/blocks/countdown/src/app.tsx

```diff
@@ -0,0 +1,220 @@
+
+import { BlockComponent } from "blockprotocol/react";
+import DatePicker from "react-datepicker";
+import {
+  isFuture,
+  isPast,
+  formatDistanceToNow,
+  formatDistanceToNowStrict,
+  formatRelative,
+  format,
+} from "date-fns";
+
+import "react-datepicker/dist/react-datepicker.css";
+
+type Range = {
+  start: Date | null;
+  end: Date | null;
+};
+
+type AppProps = {
+  interval: {
+    start?: Date | string;
+    end?: Date | string;
+  };
+  selectsRange: boolean;
+  strict: boolean;
+};
+
+// const Checkbox = ({ label, value, onChange }) => {
+//   return (
+//     <label>
+//       <input type="checkbox" checked={value} onChange={onChange} />
+//       {label}
+//     </label>
+//   );
+// };
+
+const formatDate = (date: Date, rangeSelection: boolean) => {
+  if (rangeSelection) {
+    return format(date, "PPP");
+  } else {
+    return formatRelative(date, new Date());
+  }
+};
+
+const formatRange = (range: Range) => {
+  if (range.start !== null && range.end !== null) {
+    return `${formatDate(range.start, range.end !== null)} - ${formatDate(
+      range.end,
+      range.end !== null,
+    )}`;
+  } else {
+    return "";
+  }
+};
+
+const calculateTime = (range: Range, strict: boolean) => {
+  if (range.start === null) {
+    return {
+      prefix: "",
+      offset: "Please select a date",
+      postfix: "",
+    };
+  }
+
+  let relative;
+  let date;
+  if (isFuture(range.start)) {
+    date = range.start;
+    relative = "until";
+  } else if (range.end !== null && isPast(range.end)) {
+    date = range.end;
+    relative = "since";
+  } else if (range.end === null) {
+    date = range.start;
+    relative = "since";
+  } else {
+    return {
+      prefix: "",
+      offset: formatRange(range),
+      postfix: "is now",
+    };
+  }
+
+  let distance;
+  if (strict) {
+    distance = formatDistanceToNowStrict(date);
+  } else {
+    distance = formatDistanceToNow(date, { includeSeconds: true });
+  }
+
+  return {
+    prefix: `${distance} ${relative}`,
+    offset: formatDate(date, range.end !== null),
+    postfix: "",
+  };
+};
+
+export const App: BlockComponent<AppProps> = ({
+  interval = {
+    start: null,
+    end: null,
+  },
+  selectsRange = false,
+  strict = false,
+  entityId,
+  accountId,
+  updateEntities,
+}) => {
+  const [range, setRange] = useState<Range>({
+    start: !interval.start ? null : new Date(interval.start),
+    end: !interval.end ? null : new Date(interval.end),
+  });
+  const [rangeSelection, setRangeSelection] = useState(selectsRange);
+
+  const [clock, setClock] = useState({ now: new Date() });
+  const tick = () => setClock({ now: new Date() });
+
+  useEffect(() => {
+    setRange({
+      start: !interval.start ? null : new Date(interval.start),
+      end: !interval.end ? null : new Date(interval.end),
+    });
+    setRangeSelection(selectsRange);
+  }, [interval, selectsRange, strict]);
+
+  useEffect(() => {
+    // Tick once per second
```

Comment:
This will tick at most every second, but it might take longer than a second to tick, depending on how long it takes for JavaScript to get through the other events it has to get through. Not a big deal in this scenario but worth knowing (sorry if you already did).

Reply:
I was aware of that issue, on my local machine it took about 1002 ms per update. I agree that here it doesn't really matter, but I put a comment here anyway to clarify this: addc476357ef2ec639cf27fead1d7c9a3e6ceaa9